scmailclient/scmail.py

```diff
@@ -0,0 +1,493 @@
+
+    print()
+    logging.info("Config file exists. Load Successful.")
+    return config[1]
+
+
+@click.pass_context
+def set_user_info(ctx, gnupghome):
+    """Set user information and return gnupghome and email."""
+    config = "gnupghome=" + gnupghome + "\n"
+
+    # Store config file and other information.
+    ctx.config.touch(exist_ok=False)
+    with ctx.config.open("w") as f:
+        f.write(config)
+
+    logging.info("Setting user information finish.")
+    return gnupghome
+
+
+@click.command(name="create")
+@click.option(
+    "--name",
+    "-n",
+    prompt="Enter Name",
+    default=getpass.getuser(),
+    type=str,
+    help="Generator name.",
+)
+@click.option(
+    "--email",
+    "-e",
+    prompt="Enter Email",
+    default=f"{getpass.getuser()}@scmail.dev",
+    type=str,
+    help="The user email address.",
+)
+@click.option(
+    "--key-type",
+    "-t",
+    prompt="Enter key type",
+    default="RSA",
+    type=str,
+    help="The algorithms to generate the key. ex. RSA",
+)
+@click.option(
+    "--key-length",
+    "-l",
+    prompt="Enter key length",
+    default=1024,
+    type=int,
+    help="The length of the key.",
+)
+@click.option(
+    "--expire-date",
+    "-d",
+    prompt="Enter expire date",
+    default="2y",
+    type=str,
+    help="The expire time of the key pair. 0 or empty for never expire.",
+)
+@click.password_option(
+    "--password",
+    "-p",
+    prompt="Enter private key password",
+    type=str,
+    help="Password of private key.",
+)
+@click.pass_context
+def create_key(ctx, name, email, key_type, key_length, expire_date, password):
+    """Create gnupg key pairs."""
+    logging.debug(
+        f"Information about key:\nName: {name}\nEmail: {email}\nkey type: {key_type}\nkey length: {key_length}\nexpire date: {expire_date}\n"
+    )
+
+    # Warning if key never expire and user want to continue.
+    if expire_date == 0 and not click.confirm(
+        "0 means never expire, Do you want to continue?"
+    ):
+        logging.warning("Never expire key will be created.")
+
+    key = ctx.parent.gpg.create(
+        password=password,
+        name=name,
+        email=email,
+        key_type=key_type,
+        key_length=key_length,
+        expire_date=expire_date,
+    )
+    ctx.parent.pp.pprint(key.__dict__)
+    logging.info(f"Key Creation finished.\nFingerprint is {key.fingerprint}.")
+    return key.fingerprint
+
+
+@click.command(name="list")
+@click.option(
+    "--private",
+    "-p",
+    prompt="Do you want to show the private keys?",
+    default=False,
+    type=bool,
+    is_flag=True,
+    help="Whether show the private keys",
+)
+@click.pass_context
+def list_keys(ctx, private):
+    """List and print all the keys in the gnupg home dir."""
+    keys = ctx.parent.gpg.list_keys(private)
+    key_type = "public" if private is False else "private"
+
+    logging.info(f"{len(keys)} {key_type} keys exist.")
+    ctx.parent.pp.pprint(keys.__dict__)
```

Comment:
Again, should this actually be printing everything in `keys.__dict__`?

Reply:
Thanks, it should not. This is the old code. If I change that, I will plan to print the current key information (fingerprint, keyid, type, data, expires, length, uids), and all keymaps.
@josephsPrograms Since you working on output, how do you think about this? Do you comfortable with this plan? Or if you already change some of them, just let me know.

Reply:
I would recommend printing either just `fingerprint` or nothing at all.

Reply:
OK, the client will just print the fingerprint. What I just think about is this will cause users cannot identify the difference between keys.